src/fidesapi/database/database.py

```diff
@@ -108,3 +114,23 @@ def reset_db(database_url: str) -> None:
     version = migration_context._version  # pylint: disable=protected-access
     if version.exists(connection):
         version.drop(connection)
+
+
+def get_db_health(database_url: str) -> str:
+    """Checks if the db is reachable and up to date in alembic migrations"""
+    try:
+        engine = create_engine(database_url)
+        alembic_config = get_alembic_config(database_url)
+        alembic_script_directory = script.ScriptDirectory.from_config(alembic_config)
+        with engine.begin() as conn:
+            context = migration.MigrationContext.configure(conn)
+            if (
+                context.get_current_revision()
+                != alembic_script_directory.get_current_head()
+            ):
+                return "needs migration"
+        return "healthy"
+    except Exception as error:
+        error_type = get_full_exception_name(error)
+        log.error(f"Unable to reach the database: {error_type}: {error}")
```

Comment:
We generally don't want to leak implementation details to the user, as throwing a python stack trace at them and having them figure it out won't be a good experience for them.

I'll have to dig deeper, but maybe sqlalchemy gives more specific errors? Or maybe we can parse the returned error and check for certain keywords to then construct our own cleaner error message for the user?

Reply:
or maybe this'll just get cleaned up in #563 !

Reply:
this won't throw a whole exception, just the name of the exception and whatever message comes with it. so in the case where the db container isn't up, it'll say

```
fides-fidesctl-1  | 2022-05-17 17:27:55.022 [ERROR] (main:configure_db:53): Unable to configure database: sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "fidesctl-db" to address: Name or service not known
fides-fidesctl-1  | 
fides-fidesctl-1  | (Background on this error at: http://sqlalche.me/e/14/e3q8)
```

does that seem okay?

Reply:
yep! We're following up with another issue anyway so I'm totally ok with this :)

Reply:
Okay, to be clear, the examples I gave in [the follow up issue](https://github.com/ethyca/fides/issues/563) are a result of the changes in this PR. in other words, if we're happy with this logging, then this PR as-is would cover the follow up issue too.